worker/worker.go

```diff
@@ -0,0 +1,113 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/bluesky-social/indigo/api/bsky"
+	"github.com/bluesky-social/indigo/xrpc"
+	"github.com/jackc/pgx/v5"
+	"github.com/strideynet/bsky-furry-feed/bluesky"
+	"github.com/strideynet/bsky-furry-feed/store"
+	"go.uber.org/zap"
+)
+
+func Start(
+	ctx context.Context,
+	log *zap.Logger,
+	pdsHost string,
+	bskyCredentials *bluesky.Credentials,
+	pgxStore *store.PGXStore,
+) error {
+	client, err := bluesky.ClientFromCredentials(ctx, pdsHost, bskyCredentials)
+	if err != nil {
+		return err
+	}
+	bgsClient := bluesky.BGSClient{}
+
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			task, err := pgxStore.GetNextFollowTask(ctx)
+			if errors.Is(err, pgx.ErrNoRows) {
```

Comment:
If this error isn't a ErrNoRows, we need to log err and continue.

Reply:
ah oops yes